### PR TITLE
[MONDRIAN-2562] Mondrian can't parse query correctly in pentaho7

### PR DESCRIPTION
**Pull request by Advent51**
- fixed scheme name enclosing in quotation marks

### CODE REVIEW

**Comment by Advent51** [comment]
@lucboudreau @mkambol please review

**Comment by Advent51** [comment]
@lucboudreau @mkambol please review

**Comment by Advent51** [comment]
There was wrong issue name in PR and commit message.

File: mondrian/src/main/java/mondrian/spi/Dialect.java [hunk 1]
```diff
@@ -161,6 +161,18 @@ String quoteIdentifier(
...
+     * Encloses an identifier in quotation marks appropriate for the
+     * current SQL dialect. For example, in Oracle, where the identifiers
+     * are quoted using double-quotes,
+     * <code>quoteIdentifier("schema.name")</code> yields a string
+     * containing <code>"schema"."name"</code>.
+
+     * @param name Name to be quoted.
+     * @return Quoted identifier
+     */
+    String quoteIdentifierWithInnerDots(String name);
```
Comment (lucboudreau) [inline mondrian/src/main/java/mondrian/spi/Dialect.java]: I don't like adding a new method just for this one call. The logic that decides what to do with dots should be transparent to the caller and the dialect itself should hide that logic. I recommend removing this method and handling everything in Dialect.quoteIdentifier

Comment (Advent51) [inline mondrian/src/main/java/mondrian/spi/Dialect.java]: I cannot do that without changing the method signature.

Are you sure that I need to do that?

Comment (lucboudreau) [inline mondrian/src/main/java/mondrian/spi/Dialect.java]: Not sure I understand why this requires a change in signature. Is there a reason why ``quoteIdentifier(String val, StringBuilder buf)`` isn't ok?

Comment (Advent51) [inline mondrian/src/main/java/mondrian/spi/Dialect.java]: @lucboudreau yes, there is: we need to use the logic from `quoteIdentifierWithInnerDots` ONLY for schema name, and not for the others names (like table name or table alias).
To fully understand this, you should look at the https://github.com/pentaho/mondrian/pull/865/files#diff-ea35a8727d98201e0883f976f9cf6b34R808 and http://jira.pentaho.com/browse/MONDRIAN-2471

Comment (Advent51) [inline mondrian/src/main/java/mondrian/spi/Dialect.java]: @lucboudreau

Comment (lucboudreau) [inline mondrian/src/main/java/mondrian/spi/Dialect.java]: I understand now. So should we rename the new method to something clearer, like ``quoteSchemaIdentifier()`` ?

Comment (julianhyde) [inline mondrian/src/main/java/mondrian/spi/Dialect.java]: What is a user supposed to do if their schema name contains a dot?

**Comment by Advent51** [comment]
closed as not a bug

**Pull request closed without merging**